src/dependency/UpdateDependencies.cpp

```diff
@@ -10,39 +10,93 @@ using UKControllerPlugin::Windows::WinApiInterface;
 namespace UKControllerPlugin {
     namespace Dependency {
 
+        const std::wstring dependencyListFile = L"dependencies/dependency-list.json";
+
+        /*
+            Load up a map of key -> dependency
+        */
+        std::map<std::string, nlohmann::json> LoadDependencyList(WinApiInterface& filesystem)
+        {
+            std::map<std::string, nlohmann::json> dependencies;
+            if (!filesystem.FileExists(dependencyListFile)) {
+                LogInfo("Local dependencies file not found");
+                return dependencies;
+            }
+
+            nlohmann::json dependencyList;
+            try {
+                dependencyList = nlohmann::json::parse(filesystem.ReadFromFile(dependencyListFile));
+
+                for (
+                    nlohmann::json::const_iterator it = dependencyList.cbegin();
+                    it != dependencyList.cend();
+                    ++it
+                    ) {
+                    if (!ValidDependency(*it)) {
+                        LogWarning("Invalid dependency in dependency list, skipping: " + it->dump());
+                        continue;
+                    }
+
+                    dependencies[it->at("key").get<std::string>()] = *it;
+                }
+            }
+            catch (std::exception e)

```

Comment:
Hmmm can you be more specific here? i.e. the key exception vs something else going wrong

Reply:
In theory it should only be one exception - from JSON, just decided to do a catchall on the off chance :)